### PR TITLE
Add download-autopwn proxy module

### PR DESCRIPTION
**Pull request by yungtravla**
I've been having way too much fun with this >:-)

First of all I wanna thank the [dev team](https://github.com/orgs/bettercap/people) for building this super fast, interactive and all-inclusive version of BetterCAP! :bowing_man: 

I also love the implementation of caplets, brilliant!

This module is better than my previous one, as it lets you choose exactly which files are to be pwned for **individual User-Agents**, rather than just intercepting the specified extensions for all User-Agents (if a payload existed). 

In other words, now you can decide to intercept (for example) PDF files for Android devices as well as DMG files for MacOS devices without accidentally also intercepting PDF files for MacOS devices (or vice-versa).

Hope you enjoy!

### CODE REVIEW

**Comment by evilsocket** [comment]
dude this is huge XD it'll take me a while to check it and then merge, stay tuned but *thanks* <3

**Comment by evilsocket** [comment]
omg dude i love the way you used `env` there, merging right now, this is top quality stuff :+1:

**Comment by kgretzky** [comment]
Great job! So in order to supply file that will act as a replacement for given extension, you need to replace the payload placeholder file for that specific extension?

Really nice that it targets specific OS.

**Comment by yungtravla** [comment]
@kgretzky exactly, but make sure that all the payloads are called "payload" (for example: "payload.exe")

I've uploaded 1 byte dummy files hoping it might inspire people to make new OS-specific exploits :)

**Comment by yungtravla** [comment]
@evilsocket credit goes to you and the other devs for making it!

When I saw that we can add custom variables, I was like YESS!!! :smile:

**Comment by Matrix86** [comment]
Very good stuff! Congrats! =)
